docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,490 @@
+be used to create a segmented image.
+
+**Time Complexity:** ``O(n*log(n))`` where ``n`` is the number of pixels
+
+###### Demo
+
+```julia
+julia> using TestImages, ImageSegmentation;
+julia> img = testimage("lena_gray")
+julia> function homogeneous(img)
+           min, max = extrema(img)
+           max - min < 0.2
+       end
+julia> seg = region_splitting(img, homogeneous);
+```
+
+**Original:**
+
+![Original](assets/segmentation/lena.jpg)
+
+**Segmented Image with labels replaced by their intensity means:**
+
+![SegmentedImage](assets/segmentation/lena_seg.jpg)
+
+#### Fuzzy C-means
+
+Fuzzy C-means clustering is widely used for unsupervised image segmentation. It is an
+iterative algorithm which tries to minimize the cost function:
+
+```math
+J = \displaystyle\sum_{i=1}^{N} \sum_{j=1}^{C} \delta_{ij} \| x_{i} - c_{j} \|^2
+```
+
+Unlike K-means, it allows pixels to belong to two or more clusters. It is widely used
+for medical imaging like in the soft segmentation of brain tissue model.
+
+**Time Complexity:** ``O(n*C^2*iter)`` where ``n`` is the number of pixels, ``C`` is
+number of clusters and ``iter`` is the number of iterations.
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, Images;
+julia> img = load("flower.jpg");
+julia> r = fuzzy_cmeans(img, 3, 2);
+```
+
+**Original** [(source)](https://upload.wikimedia.org/wikipedia/commons/thumb/d/d9/Flower-631765_960_720.jpg/800px-Flower-631765_960_720.jpg)
+
+
+![Original](assets/segmentation/flower.jpg)
+
+**Output with pixel intensity = cluster center intensity * membership of pixel in that class**
+
+![SegmentedImage1](assets/segmentation/flower_s1.jpg) ![SegmentedImage2](assets/segmentation/flower_s2.jpg)
+![SegmentedImage3](assets/segmentation/flower_s3.jpg)
+
+#### Watershed
+
+The watershed algorithm treats an image as a topographic surface where bright pixels correspond to peaks and dark pixels correspond to valleys. The algorithm starts flooding from valleys (local minima) of this topographic surface and region boundaries are formed when water from different sources merge. If the image is noisy, this approach leads to oversegmetation. To prevent oversegmentation, marker-based watershed is used i.e. the topographic surface is flooded from a predefined set of markers.
+
+Let's see an example on how to use watershed to segment touching objects. To use watershed, we need to modify the image such that in the new image flooding the topographic surface from the markers separates each coin. If this modified image is noisy, flooding from local minima may lead to oversegmentation and so we also need a way to find the marker positions. In this example, the inverted distance transform of the thresholded image (dist image) has the required topographic structure ([This page](https://in.mathworks.com/company/newsletters/articles/the-watershed-transform-strategies-for-image-segmentation.html) explains why this works). We can threshold the 'dist' image to get the marker positions.
```

Comment:
Add/fix backticks around `dist` (2 cases). Maybe change to a \[\`distance_transform\`\](`@ref`) link? (This will automatically link to the function reference section.)